src/User32/User32.cs

```diff
@@ -1688,14 +1692,94 @@ public static partial class User32
+        /// </summary>
+        /// <param name="lpszDeviceName">
+        /// A pointer to a null-terminated string that specifies the display device about whose graphics mode the function will obtain information.
+        /// This parameter is either <c>NULL</c> or a <see cref="DISPLAY_DEVICE.DeviceName"/> returned from <see cref="EnumDisplayDevices(string, uint, DISPLAY_DEVICE*, EnumDisplayDevicesFlags)"/>.
+        /// A <c>NULL</c> value specifies the current display device on the computer on which the calling thread is running.
+        /// </param>
+        /// <param name="iModeNum">
+        /// <para>The type of information to be retrieved. This value can be a graphics mode index or one of the following values.</para>
+        /// <para><see cref="ENUM_CURRENT_SETTINGS"/>: Retrieve the current settings for the display device.</para>
+        /// <para><see cref="ENUM_REGISTRY_SETTINGS"/>: Retrieve the settings for the display device that are currently stored in the registry.</para>
+        /// <para>
+        /// Graphics mode indexes start at zero. To obtain information for all of a display device's graphics modes, make a series of calls to <see cref="EnumDisplaySettings(string, uint, DEVMODE*)"/>, as follows: Set <paramref name="iModeNum"/> to zero for the first call,
+        /// and increment <paramref name="iModeNum"/> by one for each subsequent call.
+        /// </para>
+        /// <para>
+        /// Continue calling the function until the return value is zero.<br/>
+        /// When you call <see cref="EnumDisplaySettings(string, uint, DEVMODE*)"/> with <paramref name="iModeNum"/> set to zero, the operating system initializes and caches information about the display device.<br/>
+        /// When you call <see cref="EnumDisplaySettings(string, uint, DEVMODE*)"/> with <paramref name="iModeNum"/> set to a nonzero value,
+        /// the function returns the information that was cached the last time the function was called with iModeNum set to zero.
+        /// </para>
+        /// </param>
+        /// <param name="lpDevMode">
+        /// A pointer to a <see cref="DEVMODE"/> structure into which the function stores information about the specified graphics mode.
+        /// </param>
+        /// <remarks>
+        /// The function fails if <paramref name="iModeNum"/> is greater than the index of the display device's last graphics mode.
+        /// As noted in the description of the <paramref name="iModeNum"/> parameter, you can use this behavior to enumerate all of a display device's graphics modes.
+        /// </remarks>
+        /// <returns>If the function succeeds, the return value is nonzero. If the function fails, the return value is zero.</returns>
+        [DllImport(nameof(User32))]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern unsafe bool EnumDisplaySettings(
+            [MarshalAs(UnmanagedType.LPWStr)] string lpszDeviceName,
```

Comment:
We usually discourage `string`, preferring instead to match the header file, in this case:

```cs
[Friendly(FriendlyFlags.Array | FriendlyFlags.In)] char* lpszDeviceName,
```

At the moment, that's a little less convenient when you have a `string` you want to pass in, but #444 and #286 will take care of this.
And in particular since the input to this probably comes from `DISPLAY_DEVICE.DeviceName` which is a `char*` already, accepting it here may mean you never have to allocate a `string` in the first place.
Same for other p/invoke methods added here.

If you look, I'm sure you'll find some examples of using `string` elsewhere in this repo, including from my own contributions. Most of these were from early in the repo's development before we'd standardized on matching the header files as closely as possible.